circuit-retained/src/androidInstrumentedTest/kotlin/com/slack/circuit/retained/android/ProduceAndCollectAsRetainedStateTest.kt

```diff
@@ -0,0 +1,310 @@
+// Copyright (C) 2025 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package com.slack.circuit.retained.android
+
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.test.assertTextEquals
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.test.core.app.ActivityScenario
+import com.google.common.truth.Truth.assertThat
+import com.slack.circuit.retained.LifecycleRetainedStateRegistry
+import com.slack.circuit.retained.LocalRetainedStateRegistry
+import com.slack.circuit.retained.RetainedStateRegistry
+import com.slack.circuit.retained.produceAndCollectAsRetainedState
+import com.slack.circuit.retained.rememberRetained
+import com.slack.circuit.retained.viewModelRetainedStateRegistry
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.flow
+import leakcanary.DetectLeaksAfterTestSuccess.Companion.detectLeaksAfterTestSuccessWrapping
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+
+internal const val TAG_PRODUCED_STATE = "produced_state"
+
+/**
+ * Test class for produceAndCollectAsRetainedState function.
+ *
+ * These tests verify that:
+ * - Flow values are collected and retained across configuration changes
+ * - Changing inputs causes the producer to be rerun and state to reset
+ * - The Flow is properly scoped and doesn't cause memory leaks
+ */
+class ProduceAndCollectAsRetainedStateTest {
```

Comment:
A test we can add that covers the case for including the producer as an input would be to make a producer class that implements the `() -> Flow<T>` type, thus skipping the usual lambda memoization